projects/pastanaga/src/lib/dialog/basic-confirm-dialog.component.ts

```diff
@@ -0,0 +1,22 @@
+import { ChangeDetectionStrategy, Component, Input, OnInit, ViewChild } from '@angular/core';
+import { IDialog } from './base-dialog.component';
+import { ConfirmDialogComponent } from './confirm-dialog.component';
+
+@Component({
+    selector: 'pa-basic-confirm-dialog',
+    templateUrl: './basic-confirm-dialog.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
```

Comment:
Is this ok? As I understand, OnPush components only update when their inputs does but this components does not have any @Input :thinking:

Reply:
it allows a better management of the change detection. If you go in `BaseDialogComponent` you'll see we call `markForCheck` when needed there